src/actions/vstudio/vs2010_vcxproj.lua

```diff
@@ -1231,7 +1231,15 @@
 
 
 	function m.generateDebugInformation(cfg)
-		m.element("GenerateDebugInformation", nil, tostring(cfg.flags.Symbols ~= nil))
+		if cfg.flags.Symbols and _ACTION >= "vs2015" then
+			if cfg.flags.FastLink then
+				m.element('GenerateDebugInformation', nil, "DebugFastLink")
+			else
+				m.element('GenerateDebugInformation', nil, "Debug")
```

Comment:
I think the debugformat is very specific towards the actual format of the debug information. With symbols "off" there would be no debugformat, but with symbols "on" the format would be whatever you specified... rolling those into one API, would require every format to be rolled into the "on" and "FastLink" options, so you would get:

"Default", "DefaultC7", "Off", "On", "OnC7", "FastLink", "FastLinkC7"... I don't think that works really well, certainly not if additional formats would become available.... I'm actually surprised that only C7 is exposed in the "debugformat" API, since there is 3 other supported formats for MSVC.
- None
- Program Database (/Zi)
- Program Database for Edit And Continue (/ZI)

but interestingly to enable "EditAndContinue" you use an entirely different API
```lua
editandcontinue "On"
```